modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -642,177 +646,265 @@ public Response getEvents(@HeaderParam("Accept") String acceptHeader, @QueryPara
+            if ("identifier".equals(name)) {
+              query.withIdentifier(value);
+            } else if ("title".equals(name)) {
+              query.withTitle(value);
+            } else if ("description".equals(name)) {
+              query.withDescription(value);
+            } else if ("series_name".equals(name)) {
+              query.withSeriesName(value);
+            } else if ("language".equals(name)) {
+              query.withLanguage(value);
+            } else if ("created".equals(name)) {
+              query.withCreated(value);
+            } else if ("license".equals(name)) {
+              query.withLicense(value);
+            } else if ("rightsholder".equals(name)) {
+              query.withRights(value);
+            } else if ("is_part_of".equals(name)) {
+              query.withSeriesId(value);
+            } else if ("source".equals(name)) {
+              query.withSource(value);
+            } else if ("status".equals(name)) {
+              query.withEventStatus(value);
+            } else if ("agent_id".equals(name)) {
+              query.withAgentId(value);
+            } else if ("start".equals(name)) {
+              try {
+                Tuple<Date, Date> fromAndToCreationRange = RestUtils.getFromAndToDateRange(value);
+                query.withStartFrom(fromAndToCreationRange.getA());
+                query.withStartTo(fromAndToCreationRange.getB());
+              } catch (Exception e) {
+                return RestUtil.R
+                        .badRequest(String.format("Filter 'start' could not be parsed: %s", e.getMessage()));
+
+              }
+            } else if ("technical_start".equals(name)) {
+              try {
+                Tuple<Date, Date> fromAndToCreationRange = RestUtils.getFromAndToDateRange(value);
+                query.withTechnicalStartFrom(fromAndToCreationRange.getA());
+                query.withTechnicalStartTo(fromAndToCreationRange.getB());
+              } catch (Exception e) {
+                return RestUtil.R
+                        .badRequest(String.format("Filter 'technical_start' could not be parsed: %s", e.getMessage()));
+
+              }
+            } else {
+              logger.warn("Unknown filter criteria {}", name);
+              return RestUtil.R.badRequest(String.format("Unknown filter criterion in request: %s", name));
 
             }
-          } else if ("technical_start".equals(name)) {
-            try {
-              Tuple<Date, Date> fromAndToCreationRange = RestUtils.getFromAndToDateRange(value);
-              query.withTechnicalStartFrom(fromAndToCreationRange.getA());
-              query.withTechnicalStartTo(fromAndToCreationRange.getB());
-            } catch (Exception e) {
-              return RestUtil.R
-                      .badRequest(String.format("Filter 'technical_start' could not be parsed: %s", e.getMessage()));
+          }
+        }
 
+        if (optSort.isSome()) {
```

Comment:
I would move the sorting logic into a helper function rather than duplicating code.

Reply:
Sounds good.  Please ping me when this is done!